docs: validator recovery (#126)
axelarnetwork/axelarate-community: documentation/docs/validator_recovery.md

### Overview

This document describes the steps necessary to ensure that a validator node can be restored in case its state is lots. In order to achieve this, it is necessary that the following data is safely backed up:

* Tendermint validator key
* Axelar validator mnemonic
* Axelar proxy mnemonic
* Tofnd mnemonic

Besides the data described above, it will also be necessary to retrieve the *recovery data* associated to all the key shares that the validator was responsible for maintaining.

### Tendermint validator key

The Tendermint validator key is created when a node is launched for the first time.

> It can be found within the node's container at `/root/.axelar/config/priv_validator_key.json` (or from the node's root directory at `$ROOT_DIRECTORY/.core/config/priv_validator_key.json`).

Reviewer: I assume one of them is meant to be "host" and not node?

Author: nop, I actually mean the node's directory that can be specified in the joinTestnet script (that we do refer as root even in that context)

Its contents should look something like:

```
{
  "address": "98AF6E5D52BBB62BE6717DE8C55F16F5C013D7BE",
  "pub_key": {
    "type": "tendermint/PubKeyEd25519",
    "value": "CcspC1QDG8vz7kIW/7nPvqQ35XFJ5JLn5+li2WshP+o="
  },
  "priv_key": {
    "type": "tendermint/PrivKeyEd25519",
    "value": "VCG8TTeOSv+n9TOyy465CnUQALnoD/WJG9bloPGX0XUJyykLVAMby/PuQhb/uc++pDflcUnkkufn6WLZayE/6g=="
  }
}
```

> 

Reviewer: missing: what to do with this

Author: first I need to adjust the scripts. That is why the info is not included yet. I should have marked it as a draft PR

### Axelar mnemonics

The Axelar mnemonics are created when an node/validator is launched for the first time and subsequently outputed to the terminal.
The output looks something like:

```
**Important** write this mnemonic phrase in a safe place.
It is the only way to recover your account if you ever forget your password.

range elder logic subject never utility dutch novel sail vacuum model robust coin upper egg trophy track chimney garlic random fury laundry kiss sight
```

There should be one mnemonic for the Axelar validator key and other for the Axelar proxy key. 
The former should be displayed by the node container, while the later should be displayed by vald container.

> 

Reviewer: missing: what to do with it
add: how to get the mnemonic if you can't find it in the log (I'm pretty sure you can export the mnemonic with a command)

Author: I actually don't think you can. You can export the private key as an hex, but not the mnemonic:

```
~/scripts # axelard keys export
Error: accepts 1 arg(s), received 0
Usage:
  axelard keys export <name> [flags]

Flags:
  -h, --help            help for export
      --unarmored-hex   Export unarmored hex privkey. Requires --unsafe.
      --unsafe          Enable unsafe operations. This flag must be switched on along with all unsafe operation-specific options.

Global Flags:
      --home string              The application home directory (default "/root/.axelar")
      --keyring-backend string   Select keyring's backend (os|file|test) (default "test")
      --keyring-dir string       The client Keyring directory; if omitted, the default 'home' directory will be used
      --log_format string        The logging format (json|plain) (default "plain")
      --log_level string         The logging level (trace|debug|info|warn|error|fatal|panic) (default "info")
      --output string            Output format (text|json) (default "text")
      --trace                    print out full stack trace on errors
```

### Tofnd mnemonic

The tofnd mnemonic is created when a validator is launched for the first time.
It can be found within the tofnd container at `/.tofnd/export` (or from the node's root directory at `$ROOT_DIRECTORY/.tofnd/export`).
Its contents should look something like:

```
purchase arrow sword basic gasp category hundred town layer snow mother roast digital fragile repeat monitor wrong combine awful nature damage rib skull chalk
```

> 

Reviewer: missing: what to do with it

Author: coming with the updated scripts. But probably not today

### Recovery data

> The recovery data is stored on chain, and enables a validator to recover the shares for the keys that it took part on generating and uses to sign data.

Reviewer: ```suggestion
The recovery data is stored on chain, and enables a validator to recover key shares it created.
```

> In order to obtain this data, first it is necessary to determine which keys the validator helped generate.

Reviewer: ```suggestion
To obtain the recovery data for those key shares, you need to find out the corresponding key IDs first
```

> If you do not know which keys the validator is associatef with, attach a terminal to the node's container and perform the command:

Reviewer: ```suggestion
To query the blockchain for these key IDs, attach a terminal to the node's container and perform the command:
```

```
~/scripts # axelard q tss keySharesValidator $(axelard keys show validator --bech val -a)
- key_chain: Bitcoin
  key_id: btc-master
  key_role: KEY_ROLE_MASTER_KEY
  num_total_shares: "5"
  num_validator_shares: "1"
  snapshot_block_number: "23"
  validator_address: axelarvaloper1mx627hm02xa8m57s0xutgjchp3fjhrjwp2dw42
- key_chain: Bitcoin
  key_id: btc-secondary
  key_role: KEY_ROLE_SECONDARY_KEY
  num_total_shares: "5"
  num_validator_shares: "1"
  snapshot_block_number: "56"
  validator_address: axelarvaloper1mx627hm02xa8m57s0xutgjchp3fjhrjwp2dw4
```

> You can now levarage the information provided by the command to retrieve the recovery data for keys `btc-master` and `btc-secondary` as follows:

Reviewer: ```suggestion
In this example, the validator participated in generating the keys with ID `btc-master` and `btc-secondary`.
With the help of the key IDs, you can now retrieve the recovery data for the keys:
```

```

> axelard q tss recover $(axelard keys show validator --bech val -a) testkey btc-secondary --output json > recovery.json

Reviewer: ```suggestion
axelard q tss recover $(axelard keys show validator --bech val -a) btc-master btc-secondary --output json > recovery.json
```

```

> The command above will fetch the recovery info for the aforementioned keys and store it to the `recovery.json` file.

Reviewer: ```suggestion
The command above will fetch the recovery info for the aforementioned keys and store it to a `recovery.json` file.
```
